### DISCUSSION TITLE
AWS S3 "Unable to create directory"

### DISCUSSION THREAD

**Discussion by lostdesign** [pimcore/pimcore#8617] [Q&A / Support]
Hi everyone, I was following the [s3 docs](https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Amazon_AWS_Setup/Amazon_AWS_S3_Setup.html). When trying to upload an image, I'll be faced with the following error:

```
{"success":false,"message":"Unable to create directory: s3:\/\/1234567890-foo-bar-pimcore\/assets for asset :63"}
```

Has anyone faced the same issue? Our Bucket is only accessible by an public endpoint for displaying the images. Guess this is some relevant code - the S3Listener is currently 1:1 as per docs.

```php
<?php

use Aws\S3\S3Client;

$s3BucketName = getenv("AWS_BUCKET");

if (!empty($s3BucketName)) {
    $s3Client = new S3Client([
        'version' => 'latest',
        'region' => getenv("AWS_S3_REGION"), // choose your favorite region
        'debug' => (bool) getenv("AWS_S3_DEBUG"),
        'endpoint' => 'https://cloudfront.domain.de'
    ]);

    // Set right ACL
    $default_opts = [
        's3' => [
            'ACL' => 'private',
            'seekable' => true
        ],
    ];

    $default = stream_context_set_default($default_opts);

    $s3Client->registerStreamWrapper();

// set default file context
    \Pimcore\File::setContext(stream_context_create([
        's3' => ['seekable' => true],
    ]));
}
```

I even tried to remove any reference which could be a directory in `constants.php` -> still the error of cannot create directory. To debug further more, I installed the AWS CLI in my docker container, ran `aws s3 cp img.png s3://asdf/assets/img.png` and that worked?

**Comment by craigsmithhdmz** [comment]
I was having a similar sounding issue.  If I tried to create a folder through Pimcore it failed with permissions denied.  If I uploaded a file the file would appear in s3 as seen using a desktop app to access it, but I'd get an error that the versions folder couldn't be created which would fail the upload process.  So even though the file was in s3 it would show as a failed upload in Pimcore.  I went over the problem with our IT Director and he made some adjustments to the user permissions that allowed the system to work.  This is what he told me about the changes he made:

```
Basically, for anyone else having this problem the following has to be made:

* Make sure the user accessing the bucket (programmatically) has full permissions over the AWS resource.
* Make sure that, in the bucket settings, under bucket permissions,  all "block public access" options are set to off.

The second item is set to "on" by default
